src/patterns/ec2-app.ts

```diff
@@ -0,0 +1,105 @@
+import { HealthCheck } from "@aws-cdk/aws-autoscaling";
+import { Certificate } from "@aws-cdk/aws-certificatemanager";
+import { ApplicationProtocol, ListenerAction } from "@aws-cdk/aws-elasticloadbalancingv2";
+import { Duration } from "@aws-cdk/core";
+import { GuAutoScalingGroup, GuUserData } from "../constructs/autoscaling";
+import type { GuStack } from "../constructs/core";
+import { GuArnParameter } from "../constructs/core";
+import type { AppIdentity } from "../constructs/core/identity";
+import { GuVpc, SubnetType } from "../constructs/ec2";
+import { GuInstanceRole } from "../constructs/iam";
+import {
+  GuApplicationListener,
+  GuApplicationLoadBalancer,
+  GuApplicationTargetGroup,
+} from "../constructs/loadbalancing";
+
+interface GuEc2AppProps extends AppIdentity {
+  userData: GuUserData | string;
+  publicFacing: boolean; // could also name it `internetFacing` to match GuApplicationLoadBalancer
+  applicationPort: number;
+}
+
+interface GuMaybePortProps extends Omit<GuEc2AppProps, "applicationPort"> {
+  applicationPort?: number;
+}
+
+export enum GuApplicationPorts {
+  Node = 3000,
+  Play = 9000,
+}
+
+/*
+This pattern is under development. Please don't attempt to use it yet.
+ */
+export class GuEc2App {
+  constructor(scope: GuStack, props: GuEc2AppProps) {
+    const id = (id: string) => `${props.app}${id}`;
```

Comment:
[`AppIdentity.suffixText`](https://github.com/guardian/cdk/blob/d69bdbe181d90f0f2c502dee9b05095f1402301a/src/constructs/core/identity.ts#L16) in other places that are app aware, can this be reused here for consistency?

Reply:
Yep, done!